lib/internal/loader/ModuleJob.js

```diff
@@ -105,7 +105,6 @@ class ModuleJob {
     try {
       module.evaluate();
     } catch (e) {
-      e.stack;
```

Comment:
I believe if this is an error object, accessing the stack property ensures that the trace is generated?

You could change this to `if (e) { e.stack; }` to ensure that path is still followed.

Reply:
Node has an `isError` helper. Since `e` could be any thrown value and `.stack` could be any property with a getter it's probably better to at least have an error check just to be on the up and up.

Reply:
Can someone explain why it's important to generate the stack trace in advance?

Reply:
> Can someone explain why it's important to generate the stack trace in advance?

Could be to generate the stack before being re-thrown and adding to the stack.
Though stacks can be snipped with `Error.captureStackTrace(error, beforeFunc)`.

Reply:
> Could be to generate the stack before being re-thrown and adding to the stack.

I'm not aware that re-throwing modifies the stack. Do you have an example?

There's an old V8 bug about it: https://bugs.chromium.org/p/chromium/issues/detail?id=60240. But it was fixed in 2012.

Reply:
Here's [a similar use of `e.stack`](https://github.com/nodejs/node/blob/ddcc00bc1e8cc5815c942e1c4713ce92e7a27d40/lib/internal/loader/DefaultResolve.js#L29) with a comment suggesting the same thing. Maybe it's superstition now *(I haven't verified)*.

Reply:
cc @bmeck

Reply:
I was able to repro this a while back but cannot now. +1 to removal, but now curious about why it was showing up while working on initial loader.

Reply:
@bmeck what version of node (and thus v8) were you using when working on it initially?

Reply:
muddy to see the actual commits due to rebases, but guessing from timeline and other stuff in my fork it was probably around V8 5.4 (which was still from 2016)?

Reply:
@bmeck Did you test the syntax error case? IIRC that's the one with bad/missing stacks (`throw` and `Promise.reject` should both ensure stacks already).

Reply:
@jkrems I did not.